src/Service.Tests/Configuration/LoadConfigViaEndpoint.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Service.Controllers;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.ConfigurationEndpoints;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.TestConfigFileReader;
+
+namespace Azure.DataApiBuilder.Service.Tests.Configuration;
+
+[TestClass]
+public class LoadConfigViaEndpointTests
+{
+    [TestMethod("Testing that missing environment variables won't cause runtime failure."), TestCategory(TestCategory.COSMOSDBNOSQL)]
+    [DataRow(CONFIGURATION_ENDPOINT)]
+    [DataRow(CONFIGURATION_ENDPOINT_V2)]
+    public async Task CanLoadConfigWithMissingEnvironmentVariables(string configurationEndpoint)
+    {
+        TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+        HttpClient httpClient = server.CreateClient();
+
+        RuntimeConfig config = ReadCosmosConfigurationFromFile() with { Schema = "@env('schema')" };
+
+        ConfigurationPostParametersV2 @params = new(
```

Comment:
Good point, I forgot there were different structures between v1 and v2 endpoints and that simply adding the `DataRow` wasn't enough.

I used `@params` as I couldn't think of a better variable name than `params`, but since that's a reserved word you can use an `@` prefix to define it. You can [read it in more detail on the docs](https://learn.microsoft.com/dotnet/csharp/language-reference/language-specification/lexical-structure#643-identifiers)

Reply:
understood. I didn't doubt that it worked but a name like `configParams` would have been fine too. 

No need to change. resolving.